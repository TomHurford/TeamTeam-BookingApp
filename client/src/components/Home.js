import React, {Component} from "react";
import '../styles/Events.css';
import Event from './Events/Event';
import '../styles/Home.css';
import {getEvents} from "../utils/EventsLogic"
import search from '../utils/search.png';
import '../styles/SearchBar.css'
 

//Fetching events from the database and displaying them on the home page

class Home extends Component {
    constructor(props) {
        super(props);
        this.state = {data: [],  query: ""}
        this.fetchData();
        this.handleChange = this.handleChange.bind(this);
    }

    handleChange(event) {
        this.setState({ query: event.target.value });
    }
    
    async fetchData() {
        const events = await getEvents();
        this.setState({data: events})
    }

    welcome() {
        return(
        <div className="welcome" data-testid = "welcome-message">
            <h1>Welcome to Ticketopia!</h1>
        </div>
        )
    }

    searchBar(){
        return(
            <div className="searchBarContainer">
            {/* <img src = {search} className="searchIcon" alt="search icon" /> */}
            <input 
            className="searchBar"
            data-testid="search-bar"
            type="text" 
            placeholder = "Search for events..."
            value={this.state.query}
            onChange={this.handleChange}
            onKeyDown = {(e) => { if (e.key === 'Enter') { this.newSearch(this.state.query) }}}
            />
            </div>
        )

    }

    render(){
        return(
            <div className='page-container'>
                <div className='underlay'></div>
                <div className="homePage" data-testid = "home-component">
                    {this.welcome()}
                    {this.searchBar()}
                    <div className="events" data-testid="events-list">
                        {this.state.data.map(event => (    
                            <div className="eventCard" key={event.id} onClick={()=>this.handleClick(event.id)} >
                            <Event details={event.id} specificEvent = {event}/>
                            </div>
                        ))}
                    </div>
                </div>
            </div>
        )
    }

    handleClick= (eventId) => {
        window.location.href = '/event-details?eventId=' + eventId;
    }

    newSearch = (eventName) => {
        window.location.href = '/search-events?name=' + eventName;
    }
}
export default Home
