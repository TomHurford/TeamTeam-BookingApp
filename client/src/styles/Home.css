/*Styling for welcome message below*/

.homePage{
    display: flex;
    flex-direction: column;
    align-items: center;
    width: 100%;
    height: 100%;
    overflow-x: hidden;
    overflow-y: auto;
    
}
.events{
    margin-top: 100px;
    display: grid;
    gap: 30px;
    grid-template-columns: repeat(3, 1fr);
}

.welcome{
    font-size: 2em;
    text-align: center;
    margin-top: 50px;
    color: black;
}

.searchBar {
    width: 60%;
    max-width: 400px;
}

.searchBar > input {
    width: 100%;
    font-size: 1.3em;
    line-height: 1.5em;
    padding: 5px 15px 5px 15px;
    outline: 0;
    border: 0;
    border-bottom: 1px solid rgba(0, 0, 0, 0.4);
    border-radius: 20px;
    background-color: rgba(255, 255, 255, 0.6);
}