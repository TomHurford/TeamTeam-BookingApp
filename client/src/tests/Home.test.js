import React from 'react';
import renderer from 'react-test-renderer';
import Home from '../components/Home';
import { render, screen } from '@testing-library/react';


test('Home snapshot test', () => {
  const component = renderer.create(<Home />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('check render of search bar', () => {
    render(<Home />);
    const searchBar = screen.getByTestId('search-bar');
    expect(searchBar).toBeInTheDocument();
});

test('render Home component', () => {
    render(<Home />);
    const homeElement = screen.getByTestId('home-component');
    expect(homeElement).toBeInTheDocument();
}); 

test('check welcome message', () => {
    render(<Home />);
    const welcomeElement = screen.getByTestId('welcome-message');
    expect (welcomeElement).toHaveTextContent('Welcome to Ticketopia!');
    expect(welcomeElement).toBeInTheDocument();
});

test('list of events render on home page', () => {
    render(<Home />);
    const eventsList = screen.getByTestId('events-list');
    expect(eventsList).toBeInTheDocument();  
});


