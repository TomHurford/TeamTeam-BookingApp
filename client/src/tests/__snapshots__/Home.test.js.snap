// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home snapshot test 1`] = `
<div
  className="page-container"
>
  <div
    className="underlay"
  />
  <div
    className="homePage"
    data-testid="home-component"
  >
    <div
      className="welcome"
      data-testid="welcome-message"
    >
      <h1>
        Perfect Tickets? Perfect Time.
      </h1>
    </div>
    <div
      className="searchBarContainer"
    >
      <input
        className="searchBar"
        data-testid="search-bar"
        onChange={[Function]}
        onKeyDown={[Function]}
        placeholder="Search for events..."
        type="text"
        value=""
      />
    </div>
  </div>
</div>
`;
